packages/mui-material-next/src/InputBase/InputBase.types.ts

```diff
@@ -0,0 +1,306 @@
+  };
+  /**
+   * The components used for each slot inside the InputBase.
+   * Either a string to use a HTML element or a component.
+   * @default {}
+   */
+  slots?: InputBaseSlots;
+  /**
+   * Start `InputAdornment` for this component.
+   */
+  startAdornment?: React.ReactNode;
+  /**
+   * The system prop that allows defining system overrides as well as additional CSS styles.
+   */
+  sx?: SxProps<Theme>;
+  /**
+   * Type of the `input` element. It should be [a valid HTML5 input type](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#Form_%3Cinput%3E_types).
+   * @default 'text'
+   */
+  type?: string;
+  /**
+   * The value of the `input` element, required for a controlled component.
+   */
+  value?: unknown;
+};
+
+export interface InputBaseSlots {
+  /**
+   * The component that renders the root.
+   * @default 'div'
+   */
+  root?: React.ElementType;
+  /**
+   * The component that renders the input.
+   * @default 'input'
+   */
+  input?: React.ElementType;
+  /**
+   * The component that renders the textarea.
+   * @default 'textarea'
+   */
+  textarea?: React.ElementType;
+}
+
+export interface InputBaseTypeMap<
+  AdditionalProps = {},
+  RootComponentType extends React.ElementType = 'div',
+> {
+  props: InputBaseOwnProps & AdditionalProps;
+  defaultComponent: RootComponentType;
+}
+
+export type InputBaseProps<
+  RootComponentType extends React.ElementType = InputBaseTypeMap['defaultComponent'],
+  AdditionalProps = {},
+> = OverrideProps<InputBaseTypeMap<AdditionalProps, RootComponentType>, RootComponentType>;
+
+export type InputBaseOwnerState = Simplify<
+  InputBaseOwnProps & {
+    formControl: FormControlState | undefined;
+    hiddenLabel?: boolean;
+    focused: boolean;
+    type: React.InputHTMLAttributes<HTMLInputElement>['type'] | undefined;
+  }
+>;
+
+export type InputBaseRootSlotProps = Simplify<
+  UseInputRootSlotProps & {
+    ownerState: InputBaseOwnerState;
+    className?: string;
+    children?: React.ReactNode;
+    ref?: React.Ref<HTMLDivElement>;
+  }
+>;
+
+export type InputBaseInputSlotProps = Simplify<
+  Omit<UseInputRootSlotProps, 'onClick'> & {
+    'aria-describedby': React.AriaAttributes['aria-describedby'];
+    'aria-label': React.AriaAttributes['aria-label'];
+    'aria-labelledby': React.AriaAttributes['aria-labelledby'];
+    autoComplete: string | undefined;
+    autoFocus: boolean | undefined;
+    className?: string;
+    id: string | undefined;
+    name: string | undefined;
+    onKeyDown: React.KeyboardEventHandler<HTMLInputElement> | undefined;
+    onKeyUp: React.KeyboardEventHandler<HTMLInputElement> | undefined;
+    ownerState: InputBaseOwnerState;
+    placeholder: string | undefined;
+    readOnly: boolean | undefined;
+    ref: React.Ref<HTMLInputElement>;
+    type: React.HTMLInputTypeAttribute | undefined;
+  }
+>;
+
+declare const InputBase: OverridableComponent<InputBaseTypeMap>;
+
+export default InputBase;
```

Comment:
As the `InputBase` component file is `tsx` now, we can remove these and declare this type in the component file, for example: https://github.com/mui/material-ui/blob/master/packages/mui-joy/src/Input/Input.tsx#L345